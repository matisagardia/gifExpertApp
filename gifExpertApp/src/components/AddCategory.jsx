import { useState } from "react"

export const AddCategory = ({onNewCategory}) => {

    const [inputvalue, setInputvalue] = useState('');

    const onInputChange = (e) => setInputvalue(e.target.value);

    const onSubmit = e => {
        e.preventDefault();
        if (inputvalue.trim().length < 1) return;
        // uso callback function de setCategories
        onNewCategory(inputvalue.trim());
        setInputvalue('');
    };

  return (
    <form onSubmit={onSubmit}>
    <input type="text" placeholder="Buscar GIFs" value={inputvalue} onChange={onInputChange}/>
    </form>
  )
}
