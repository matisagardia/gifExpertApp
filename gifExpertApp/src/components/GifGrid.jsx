
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';


export const GifGrid = ({cat}) => {

    const {images, isLoading} = useFetchGifs(cat);

  return (
    <>
        <h3>{cat}</h3>

        {
            isLoading && (<h2>Loading...</h2>)
        }

        <div className='card-grid'>
            {images.map((img) => (
            <GifGridItem 
            key={img.id}
            title={img.title}
            url={img.url}
            //otra forma de traer las propiedades de img es usar el spread operator de la siguiente manera
            // {...img} de esta forma desestructuro todas las propiedades y luego en el componente puedo acceder facilmente
            />
            ))
            }
        </div>
    </>
  )
}
