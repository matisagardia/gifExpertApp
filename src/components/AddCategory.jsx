import { useState } from "react"
import {PropTypes} from 'prop-types'

export const AddCategory = ({onNewCategory}) => {

    const [inputvalue, setInputvalue] = useState('');

    const onInputChange = (e) => setInputvalue(e.target.value);

    const onSubmit = e => {
        e.preventDefault();
        if (inputvalue.trim().length < 1) return;
        onNewCategory(inputvalue.trim());
        setInputvalue('');
    };

  return (
    <form onSubmit={onSubmit} aria-label='form'>
    <input type="text" placeholder="Buscar GIFs" value={inputvalue} onChange={onInputChange}/>
    </form>
  )
}


AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}