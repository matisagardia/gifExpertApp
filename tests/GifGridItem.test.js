import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../src/components/GifGridItem";

describe('GifGridItem Test', () => { 

    const title = 'Saitama';
    const url = 'https://pruebaurl.com';

    test('should pass props', () => { 

        const {container} = render(<GifGridItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();

     });
     
     test('should return img with url and alt', () => { 

        render(<GifGridItem title={title} url={url}/>);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);

     });

     test('should show the title in the component', () => { 

        render(<GifGridItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy;

     });

 });