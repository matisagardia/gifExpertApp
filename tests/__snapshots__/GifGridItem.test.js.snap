// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem Test should pass props 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://pruebaurl.com"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
